File: src/main/java/com/tkj/garagedoorz/domain/DoorStatus.java
package com.tkj.garagedoorz.domain;

/**
 * DoorStatus model represents the state of a garage door.
 * @author Thomas G. Kenny Jr.
 */
public class DoorStatus {

	final String STATUS_CLOSED = "closed";
	final String STATUS_CLOSING = "closing";
	final String STATUS_OPEN = "open";
	final String STATUS_OPENING = "opening";
	
    final private String name;
    final private boolean isClosed;

    public DoorStatus( final String doorName, final boolean isDoorClosed ) {

        name = doorName;
        isClosed = isDoorClosed;

    }

    public String getDoorName() {
        return name;
    }

    public boolean isDoorClosed() {
        return isClosed;
    }
    
    public String getStatus() {
    	if(isClosed) return STATUS_CLOSED;
    	return STATUS_OPEN;
    }

}

File: src/main/java/com/tkj/garagedoorz/domain/DoorStatus.kt
package com.tkj.garagedoorz.domain

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * DoorStatus model represents the state of a garage door.
 * @author Thomas G. Kenny Jr.
 */
data class DoorStatus( @JsonProperty( "doorName" ) val doorName: String, val isDoorClosed: Boolean ) {

    private val STATUS_CLOSED = "closed"
    private val STATUS_OPEN = "open"

    @JsonProperty("status")
    fun status(): String {

        return if ( isDoorClosed ) STATUS_CLOSED else STATUS_OPEN
    }

    override fun equals( other: Any? ): Boolean {
        if( this === other ) return true
        if( other !is DoorStatus ) return false

        if( doorName != other.doorName ) return false

        return true
    }

    override fun hashCode(): Int {

        return doorName.hashCode()
    }

    override fun toString(): String {
        return "DoorStatus(doorName='$doorName', doorStatus=$isDoorClosed, status='${status()}')"
    }

}
